### PR TITLE
Add shell EMR Kick-off job Lambda

### PR DESCRIPTION
**Pull request by ciaranevans**
* Adds Lambda function with a dummy zip, ready for the real artifact to replace it
* Integrates the Lambda function with Cloudwatch logs
* Refactored main.tf to include modules (buckets + emr_lambda) to separate concerns better

pbi: 146656

### CODE REVIEW

File: src/main/terraform/infra_deployment/main.tf [hunk 1]
```diff
@@ -7,38 +7,19 @@ terraform {
...
-
-resource "aws_ssm_parameter" "heatmap_bucket" {
-  name = "/buckets/heatmap_bucket"
-  type = "SecureString"
-  value = "${aws_s3_bucket.heatmap_bucket.bucket}"
-}
+module "emr_lambda" {
+  source = "./modules/emr_lambda"
+  component_bucket = "${module.s3_buckets.component_bucket}"
+}
```
Comment (andyphelps) [inline src/main/terraform/infra_deployment/main.tf]: Newline

File: src/main/terraform/infra_deployment/modules/emr_lambda/main.tf [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  source = "${path.module}/emr_lambda.zip"
+}
+
+resource "aws_lambda_function" "emr_lambda" {
+  s3_bucket = "${var.component_bucket}"
+  s3_key = "${aws_s3_bucket_object.emr_lambda.key}"
+  function_name    = "emr_orchestration"
+  role             = "${aws_iam_role.iam_for_emr_lambda.arn}"
+  handler          = "index.handler"
+  runtime          = "nodejs8.10"
```
Comment (polmacaoidh) [inline src/main/terraform/infra_deployment/modules/emr_lambda/main.tf]: Think we should decide upon a version of node.js, perhaps a sepcific 10.x?

Comment (andyphelps) [reply src/main/terraform/infra_deployment/modules/emr_lambda/main.tf]: 8.10 to be used

File: .circleci/config.yml [hunk 3]
```diff
@@ -31,3 +48,6 @@ workflows:
       - build
       - check
       - test
+  deploy_infra:
+    jobs:
+      - deploy-infra
```
Comment (eganjs) [inline .circleci/config.yml]: filter to master branch https://circleci.com/docs/2.0/configuration-reference/#example-8

Comment (andyphelps) [reply .circleci/config.yml]: Done

File: src/main/terraform/infra_deployment/modules/emr_lambda/main.tf [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_function" "emr_lambda" {
+  filename      = "${path.module}/emr_dummy.zip"
```
Comment (JoelClemence) [inline src/main/terraform/infra_deployment/modules/emr_lambda/main.tf]: Name of filename?

File: README.md [hunk 5]
```diff
@@ -3,3 +3,21 @@
 [![CircleCI](https://circleci.com/gh/UKHO/spark-ais-to-raster.svg?style=svg)](https://circleci.com/gh/UKHO/spark-ais-to-raster)
 
 This is a Spark job to convert AIS data into a GeoTIFF heatmap. It is intended to be run on AWS EMR.
+
+Lambda aws cli update code example:
+aws lambda update-function-code --zip-file fileb://emr_lambda.zip --function-name emr_orchestration
+
+
+CircleCI requires an AWS IAM to run the terraform so create an IAM called circleci and give it the following permissions:
```
[the remark this answers is not in the document]
Comment (andyphelps) [reply README.md]: Don't know.  Think for now it's a user.  We can extend as a role when needed

File: src/main/terraform/infra_deployment/decrypt_parameters.sh [hunk 6]
```diff
@@ -5,7 +5,7 @@ echo "This is stored in AWS SSM under 'ais_to_heatmaps_parameters_password'"
 
 read -s AIS_TO_HEATMAPS_PARAMETERS_PASSWORD
 
-openssl enc -aes-256-cbc -d -k $AIS_TO_HEATMAPS_PARAMETERS_PASSWORD -in parameters.json.enc -out parameters.secret.json
+openssl enc -aes-256-cbc -md md5 -d -k $AIS_TO_HEATMAPS_PARAMETERS_PASSWORD -in parameters.json.enc -out parameters.secret.json
```
[the remark this answers is not in the document]
Comment (andyphelps) [reply src/main/terraform/infra_deployment/decrypt_parameters.sh]: git-secret is nasty to install apparently.

File: README.md [hunk 7]
```diff
@@ -3,3 +3,21 @@
 [![CircleCI](https://circleci.com/gh/UKHO/spark-ais-to-raster.svg?style=svg)](https://circleci.com/gh/UKHO/spark-ais-to-raster)
 
 This is a Spark job to convert AIS data into a GeoTIFF heatmap. It is intended to be run on AWS EMR.
+
+Lambda aws cli update code example:
+aws lambda update-function-code --zip-file fileb://emr_lambda.zip --function-name emr_orchestration
```
[the remark this answers is not in the document]
Comment (andyphelps) [reply README.md]: Definatley need to re-visit this later but is acceptable for now.

File: src/main/terraform/infra_deployment/modules/emr_lambda/main.tf (hunk 4 above)
Comment (andyphelps) [reply src/main/terraform/infra_deployment/modules/emr_lambda/main.tf]: Is a dummy (empty in fact) jar.  aws cli or circleci orb will upload the real code.

**Pull request merged**